src/app/Routes.tsx

```diff
@@ -64,101 +65,103 @@ const Routes = (): JSX.Element => (
       <Route
         key={path}
         path={path}
-        render={() => (
+        element={
           <ErrorBoundary>
             <Controllers />
           </ErrorBoundary>
-        )}
+        }
       />
     ))}
-    {[devicesURLs.devices.index, devicesURLs.device.index(null, true)].map(
-      (path) => (
-        <Route
-          key={path}
-          path={path}
-          render={() => (
-            <ErrorBoundary>
-              <Devices />
-            </ErrorBoundary>
-          )}
-        />
-      )
-    )}
+    <Route
+      path={devicesURLs.devices.index}
+      element={
+        <ErrorBoundary>
+          <DeviceList />
+        </ErrorBoundary>
+      }
+    />
+    <Route
+      path={`${devicesURLs.device.index(null, true)}/*`}
```

Comment:
Only the index needs to be defined at the top level, all the child routes are handled inside the details component as before.